helion/_compiler/compile_environment.py

```diff
@@ -68,6 +69,8 @@ def __init__(self, device: torch.device, settings: Settings) -> None:
             collections.Counter()
         )
 
+        self.symint_to_grid_index_type: dict[str, GridIndexType] = {}
```

Comment:
Could we implement this as a new subclass of tiling (similar to how reductions are a subclass of tiling)?